### PR TITLE
Personal/namalu/branch specific deploy

### PR DESCRIPTION
**Pull request by namalu**
This PR updates the `ContainerApp-SingleAzureDeploy.bicep` to provide a way to specify a specific GitHub url (for cases where a user may want to deploy from a fork) and/or a specific branch (this can be used by users and pipelines).

### CODE REVIEW

File: deploy/templates/bicep/UploadTemplates.bicep [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@minLength(3)
+@minLength(6)
 @maxLength(16)
 @description('Basename that is used to name provisioned resources. Should be alphanumeric, at least 3 characters and less than 16 characters.')
```
Comment (rogordon01) [inline deploy/templates/bicep/UploadTemplates.bicep:3]: Nit:  Can we update the description as well to reflect the new `minLength`?

File: deploy/templates/bicep/ContainerAppSetup.bicep [hunk 2]
```diff
@@ -1,4 +1,4 @@
-@minLength(3)
+@minLength(6)
 @maxLength(16)
 @description('Basename that is used to name provisioned resources. Should be alphanumeric, at least 3 characters and less than 16 characters.')
```
Comment (rogordon01) [inline deploy/templates/bicep/ContainerAppSetup.bicep:3]: Same comment as below. I think it'd need to be done for all bicep files

**Review by rogordon01 — CHANGES_REQUESTED** [review]

**Review by rogordon01 — APPROVED** [review]

**Review by dustinburson — DISMISSED** [review]

File: deploy/templates/bicep/UploadTemplates.bicep (hunk 1 above)
Comment (namalu) [reply deploy/templates/bicep/UploadTemplates.bicep:3]: Will do.

File: deploy/templates/bicep/ContainerAppSetup.bicep (hunk 2 above)
Comment (namalu) [reply deploy/templates/bicep/ContainerAppSetup.bicep:3]: Will do.

**Review by rogordon01 — APPROVED** [review]

**Pull request merged**